Signup-Form/src/components/Signup.tsx

```diff
@@ -0,0 +1,272 @@
+      );
+      const countriesData = await response.json();
+
+      setCountries(countriesData);
+    }
+
+    fetchCountries();
+  }, []);
+  const emailRegex: RegExp =
+    /^[ ]*([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})[ ]*$/i;
+  const passwordRegex: RegExp = /^[a-zA-Z\!@#\$%\^&\*-]{8,}$/;
+  const [errors, setErrors] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    country: "",
+    city: "",
+  });
+
+  const validateValues = () => {
+    let foundErrors = {
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      country: "",
+      city: "",
+    };
+    if (formData.firstName.length == 0) {
+      foundErrors.firstName = "First Name is required.";
+    } else if (
+      formData.firstName.length < 4 ||
+      formData.firstName.length > 25
+    ) {
+      foundErrors.firstName =
+        "First Name should be between 4 and 25 characters.";
+    }
+    if (formData.lastName.length == 0) {
+      foundErrors.lastName = "Last Name is required.";
+    } else if (formData.lastName.length < 4 || formData.lastName.length > 25) {
+      foundErrors.lastName = "Last Name should be between 4 and 25 characters.";
+    }
+
+    if (formData.email.length == 0) {
+      foundErrors.email = "Email is required.";
+    } else if (!formData.email.match(emailRegex)) {
+      foundErrors.email = "Email format is invalid.";
+    }
+    if (formData.password.length == 0) {
+      foundErrors.password = "Password is required.";
+    } else if (!formData.password.match(passwordRegex)) {
+      foundErrors.password =
+        "Password must contain minimum 8 characters and at least one special character.";
+    }
+    if (formData.confirmPassword.length == 0) {
+      foundErrors.confirmPassword = "Please confirm password.";
+    } else if (formData.confirmPassword !== formData.password) {
+      foundErrors.confirmPassword = "Password does't match.";
+    }
+    if (formData.country.length == 0) {
+      foundErrors.country = "Please select a country.";
+    }
+    if (formData.city.length == 0) {
+      foundErrors.city = "Please select a city.";
+    }
+    setErrors(foundErrors);
+
+    const isValid = (
+      Object.keys(foundErrors) as [
+        "firstName",
+        "lastName",
+        "email",
+        "password",
+        "confirmPassword",
+        "country",
+        "city"
+      ]
+    ).every((key) => foundErrors[key].length == 0);
+
+    return isValid;
+  };
+
+  const onChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    setFormData({ ...formData, [event.target.name]: event.target.value });
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isValid = validateValues();
+    if (isValid) {
+      localStorage.setItem("email", formData.email);
+      navigate("/login");
+    }
```

Comment:
Aici ar fi trebuit setat tot user-ul in localstorage (facut push intr-un array de users).
Iar codul de la pagina de la login ar trebui sa foloseasca doar userii din localstorage care au facut sign up.

In rest arata totul bine :D Congrats

```suggestion
    if (isValid) {
      const users = JSON.parse(localStorage.getItem("users"));
      if(!users) {
          localStorage.setItem("users", JSON.stringify([{ email, password }]);
      } else {
          users.push({ email, password });
      }
      navigate("/login");
    }
```